ci/udev_check.c

```diff
@@ -0,0 +1,34 @@
+#include <libudev.h>
+#include <stdio.h>
+
+int main(int argc, char **argv)
+{
+        char *device;
+        struct udev *udev;
+        struct udev_device *dev = NULL;
+
+        if ((udev = udev_new()) == NULL) {
+                printf("failed to get udev device\n");
+                return (-1);
+        }
+
+        if (argc > 1) {
+                device = argv[1];
+        } else {
+                fprintf(stderr, "device is not provided\n");
+                return (-1);
+        }
+
+        dev = udev_device_new_from_subsystem_sysname(udev, "block", device);
+
+        if ((dev != NULL) && udev_device_get_is_initialized(dev)) {
+                printf("device = %s is initialized by udev\n", device);
+        } else {
+                printf("device = %s is not initialized by udev\n", device);
```

Comment:
modify this code and return non zero in this case, and use the return code to check for failure/success of this command.

Reply:
Since we are executing this via `kubectl exec -it ...` since the process execution i.e udev_check sda is executed successfully it returns 0 even if the process returns -1.

Reply:
@mittachaitu
```
 $ kubectl exec -it openebs-ndm-dzzpk -n openebs -- ls abc
ls: cannot access 'abc': No such file or directory
command terminated with exit code 2
k8s@e2e1-master:~$ echo $?
2
```

Reply:
let's print `errno` also

Reply:
> @mittachaitu
> 
> ```
>  $ kubectl exec -it openebs-ndm-dzzpk -n openebs -- ls abc
> ls: cannot access 'abc': No such file or directory
> command terminated with exit code 2
> k8s@e2e1-master:~$ echo $?
> 2
> ```

I might try in older versions of Kubernetes where it returned 0 .

Reply:
Done